File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/QuestionPaymentEntity.java
package com.eager.questioncloud.core.domain.payment;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "question_payment")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class QuestionPaymentEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String paymentId;

    @Column
    private Long userId;

    @Column
    private Long userCouponId;

    @Column
    private int amount;

    @Column
    @Enumerated(EnumType.STRING)
    private QuestionPaymentStatus status;

    @Column
    private LocalDateTime createdAt;

    @Builder
    public QuestionPaymentEntity(Long id, String paymentId, Long userId, Long userCouponId, int amount, QuestionPaymentStatus status,
        LocalDateTime createdAt) {
        this.id = id;
        this.paymentId = paymentId;
        this.userId = userId;
        this.userCouponId = userCouponId;
        this.amount = amount;
        this.status = status;
        this.createdAt = createdAt;
    }

    public QuestionPayment toModel() {
        return QuestionPayment.builder()
            .id(id)
            .paymentId(paymentId)
            .userId(userId)
            .userCouponId(userCouponId)
            .amount(amount)
            .status(status)
            .createdAt(createdAt)
            .build();
    }

    public static QuestionPaymentEntity from(QuestionPayment questionPayment) {
        return QuestionPaymentEntity.builder()
            .id(questionPayment.getId())
            .paymentId(questionPayment.getPaymentId())
            .userId(questionPayment.getUserId())
            .userCouponId(questionPayment.getUserCouponId())
            .amount(questionPayment.getAmount())
            .status(questionPayment.getStatus())
            .createdAt(questionPayment.getCreatedAt())
            .build();
    }
}

File: qc-core/qc-pay/qc-pay-rdb/src/main/kotlin/com/eager/questioncloud/payment/entity/QuestionPaymentEntity.kt
package com.eager.questioncloud.payment.entity

import com.eager.questioncloud.payment.domain.QuestionPayment
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "question_payment")
class QuestionPaymentEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val paymentId: Long = 0,
    @Column val orderId: String,
    @Column val userId: Long,
    @Column val realAmount: Int,
    @Column val originalAmount: Int,
    @Column val createdAt: LocalDateTime,
) {
    companion object {
        fun createNewEntity(questionPayment: QuestionPayment): QuestionPaymentEntity {
            return QuestionPaymentEntity(
                orderId = questionPayment.order.orderId,
                userId = questionPayment.userId,
                realAmount = questionPayment.realAmount,
                originalAmount = questionPayment.originalAmount,
                createdAt = questionPayment.createdAt,
            )
        }
    }
}
